### PR TITLE
Disallow assignment to constants and expressions

### LINKED ISSUE #6221: Constants are not very constant
**Operating system or device - Godot version:**
Windows 10 - v2.1.stable.official on Steam

**Issue description** (what happened, and what was expected):
I'm not sure if this is intentional or not since I can't find any documentation on it. I'm able to modify the value of constants after they are declared, including the [@Gloabl Scope](http://docs.godotengine.org/en/latest/classes/class_@global%20scope.html) constants. Usually in other languages _constants aren't variables and may not be modified_.

**Steps to reproduce:**
Try this in any function that will get called:

```
print(TYPE_BOOL)
TYPE_BOOL = 999
print(TYPE_BOOL)
```

Output:

> 1
> 999

You can even try declaring your own constants.

### LINKED ISSUE #6824: GDScript doesn't throw errors for assigning to unassignable expressions
**Operating system or device - Godot version:**
Win10 Home 64Bit - Godot v.2.2.custom_build

**Issue description** (what happened, and what was expected):
No Errorhandling for "bad" Syntax for (in editor and cmd.exe window):

  get_node("MyArray") = add_child(myMap[x][y],false)
  add_child(myMap[x][y],false)

**Steps to reproduce:**
see my codessnippet : http://pastebin.com/E7ApcFxj

**Link to minimal example project** (optional but very welcome):
see above 🎃

### PR DESCRIPTION
**Pull request by bojidar-bg**
Fixes #6221, fixes #6824

### CODE REVIEW

File: modules/gdscript/gd_parser.cpp [hunk 1]
```diff
@@ -2014,7 +2023,7 @@ void GDParser::_parse_block(BlockNode *p_block,bool p_static) {
 			} break;
 			default: {
 
-				Node *expression = _parse_and_reduce_expression(p_block,p_static,false,true);
+				Node *expression = _parse_and_reduce_expression(p_block,p_static,true,true);
```
Comment (bojidar-bg) [inline modules/gdscript/gd_parser.cpp]: I'm not really sure why this was left to false (it is `p_reduce_const`), would be nice if @reduz confirms :smiley:

Comment (akien-mga) [inline modules/gdscript/gd_parser.cpp]: I would say copy paste mistake :)

Comment (bojidar-bg) [inline modules/gdscript/gd_parser.cpp]: @akien-mga You think so? It is never true in the other parts of the code :laughing: 

Let's see if we break anything though...

Comment (volzhs) [inline modules/gdscript/gd_parser.cpp]: turning back to false fixes #7412

**Pull request merged**